pkg/policy/mapstate.go

```diff
@@ -2049,15 +2084,21 @@ func (mc *MapChanges) consumeMapChanges(policyOwner PolicyOwner, policyMapState
 			// insert but do not allow non-redirect entries to overwrite a redirect entry,
 			// nor allow non-deny entries to overwrite deny entries.
 			// Collect the incremental changes to the overall state in 'mc.adds' and 'mc.deletes'.
-			policyMapState.denyPreferredInsertWithChanges(key, entry, identities, features, changes)
+			p.policyMapState.denyPreferredInsertWithChanges(key, entry, p.SelectorCache, features, changes)
 		} else {
 			// Delete the contribution of this cs to the key and collect incremental changes
-			for cs := range mc.changes[i].Value.owners { // get the sole selector
-				policyMapState.deleteKeyWithChanges(mc.changes[i].Key, cs, identities, changes)
+			for cs := range mc.synced[i].Value.owners { // get the sole selector
+				p.policyMapState.deleteKeyWithChanges(mc.synced[i].Key, cs, p.SelectorCache, changes)
 			}
 		}
 	}
-	mc.changes = nil
+
+	// move version to the caller
+	version := mc.version
+	mc.version = nil
+
+	mc.synced = nil
 	mc.mutex.Unlock()
```

Comment:
nit: Same here, large function, easy to add an early return and leaking the lock by accident

Reply:
Changed to use `defer` :-)